itest/rust/src/engine_tests/native_audio_structures_test.rs

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::thread;
+use std::time::Duration;
+
+use godot::builtin::Vector2;
+use godot::classes::native::AudioFrame;
+use godot::classes::{
+    AudioEffect, AudioEffectInstance, AudioServer, AudioStreamGenerator,
+    AudioStreamGeneratorPlayback, AudioStreamPlayer, Engine, IAudioEffect, IAudioEffectInstance,
+    SceneTree,
+};
+use godot::global::godot_print;
+use godot::obj::{Base, Gd, NewAlloc, NewGd};
+use godot::register::{godot_api, GodotClass};
+
+use crate::framework::itest;
+
+#[derive(GodotClass)]
+#[class(base = AudioEffect, init)]
+struct AudioEffectReceiver {
+    base: Base<AudioEffect>,
+}
+
+#[godot_api]
+impl IAudioEffect for AudioEffectReceiver {
+    fn instantiate(&mut self) -> Option<Gd<AudioEffectInstance>> {
+        Some(AudioEffectReceiverInstance::new_gd().upcast())
+    }
+}
+
+#[derive(GodotClass)]
+#[class(base = AudioEffectInstance, init)]
+struct AudioEffectReceiverInstance {
+    was_called: bool,
+    base: Base<AudioEffectInstance>,
+}
+
+#[godot_api]
+impl IAudioEffectInstance for AudioEffectReceiverInstance {
+    unsafe fn process(
+        &mut self,
+        _src_buffer: *const std::ffi::c_void,
+        dst_buffer: *mut AudioFrame,
+        _frame_count: i32,
+    ) {
+        (*dst_buffer).left = 15.0;
+        (*dst_buffer).right = -12.0;
+        self.was_called = true;
+    }
+}
+
+#[derive(GodotClass)]
+#[class(base = AudioEffect, init)]
+struct AudioEffectAsserter {
+    base: Base<AudioEffect>,
+}
+
+#[godot_api]
+impl IAudioEffect for AudioEffectAsserter {
+    fn instantiate(&mut self) -> Option<Gd<AudioEffectInstance>> {
+        Some(AudioEffectAsserterInstance::new_gd().upcast())
+    }
+}
+
+#[derive(GodotClass)]
+#[class(base = AudioEffectInstance, init)]
+struct AudioEffectAsserterInstance {
+    was_called: bool,
+    base: Base<AudioEffectInstance>,
+}
+
+#[godot_api]
+impl IAudioEffectInstance for AudioEffectAsserterInstance {
+    unsafe fn process(
+        &mut self,
+        src_buffer: *const std::ffi::c_void,
+        _dst_buffer: *mut AudioFrame,
+        _frame_count: i32,
+    ) {
+        let src = src_buffer as *const AudioFrame;
+
+        assert_eq!((*src).left, 15.0);
+        assert_eq!((*src).right, -12.0);
+
+        self.was_called = true;
+    }
+}
+
+#[itest]
+fn native_audio_structure_out_parameter() {
+    // create two audio effects: one that writes to the out parameter and one that reads the incomming parameter.
```

Comment:
```suggestion
    // Create two audio effects: one that writes to the out parameter and one that reads the incomming parameter.
```
Please everywhere capitalization and periods at end.